sigstore-java/src/main/java/dev/sigstore/tuf/TufClient.java

```diff
@@ -0,0 +1,279 @@
+              .createRequestFactory(
+                  request -> {
+                    request.setParser(GsonFactory.getDefaultInstance().createJsonObjectParser());
+                  })
+              .buildGetRequest(nextVersionUrl);
+      req.getHeaders().setAccept("application/json; api-version=2.0");
+      req.getHeaders().setContentType("application/json");
+      req.setThrowExceptionOnExecuteError(false);
+      var resp = req.execute();
+      if (resp.getStatusCode() == 404) {
+        // No newer versions, go to 5.3.10.
+        break;
+      }
+      if (resp.getStatusCode() != 200) {
+        throw new TufException(
+            String.format(
+                "Unexpected return from mirror. Status code: %s, status message: %s"
+                    + resp.getStatusCode()
+                    + resp.getStatusMessage()));
+      }
+      byte[] rootBytes = resp.getContent().readNBytes(MAX_META_BYTES);
+      if (rootBytes.length == MAX_META_BYTES && resp.getContent().read() != -1) {
+        throw new MetaFileExceedsMaxException(nextVersionUrl.toString(), MAX_META_BYTES);
+      }
+      var newRoot = GSON.get().fromJson(new String(rootBytes, StandardCharsets.UTF_8), Root.class);
+
+      // 5.3.4) we have a valid next version of the root.json. Check that the file has been signed
+      // by:
+      //   a) a threshold (from step 2) of keys specified in the trusted metadata
+      //   b) and a threshold of keys in the new root.json.
+      //    Fail if either a or b aren't true.
+      var trustedRootKeys = trustedRoot.getSignedMeta().getKeys();
+      var newRootSignatures = newRoot.getSignatures();
+      byte[] newRootMetaBytes = newRoot.getCanonicalSignedBytes();
+      // Verify our new root meta against the trusted root keys.
+      RootRole trustedRootRoleMeta = trustedRoot.getSignedMeta().getRole(Role.Name.ROOT);
+      verifyDelegate(
+          newRootSignatures, trustedRootRoleMeta.getThreshold(), trustedRootKeys, newRootMetaBytes);
+
+      var newRootRoleMeta = newRoot.getSignedMeta().getRole(Role.Name.ROOT);
+      var newRootKeys = newRoot.getSignedMeta().getKeys();
+      // Verify our new root meta against the new root keys.
+      verifyDelegate(
+          newRootSignatures, newRootRoleMeta.getThreshold(), newRootKeys, newRootMetaBytes);
+
+      // 5.3.5) We've taken the liberty to modify 5.3.5 to just validate that the new root meta
+      // matches the version we pulled based off of the pattern {version}.root.json. We know due to
+      // the
+      // loop constraints that it is larger than the current version.
+      if (newRoot.getSignedMeta().getVersion() != nextVersion) {
+        throw new RoleVersionException(nextVersion, newRoot.getSignedMeta().getVersion());
+      }
+      // 5.3.7) set the trusted root metadata to the new root
+      trustedRoot = newRoot;
+      // 5.3.8) persist to repo
+      Path localTrustRoot = localStore.resolve("root.json");
```

Comment:
do we not care about the numbered roots naming convention anymore?

Reply:
we're storing the current up-to-date trust root as root.json.  It's only on the mirror that each version is stored as {version}.root.json.